service/app/utils/module/RequestInterceptor.java

```diff
@@ -0,0 +1,92 @@
+package utils.module;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sunbird.auth.verifier.ManagedTokenValidator;
+import org.sunbird.request.HeaderParam;
+import org.sunbird.util.JsonKey;
+import play.mvc.Http;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Request interceptor responsible to authenticated HTTP requests
+ *
+ * @author Amit Kumar
+ */
+public class RequestInterceptor {
+  static Logger logger = LoggerFactory.getLogger(RequestInterceptor.class);
+
+  private RequestInterceptor() {}
+
+  private static String getUserRequestedFor(Http.Request request) {
+    String requestedForUserID = null;
+    JsonNode jsonBody = request.body().asJson();
+    if (!(jsonBody == null)) { // for search and update and create_mui api's
+      if (!(jsonBody.get(JsonKey.REQUEST).get(JsonKey.USER_ID) == null)) {
+        requestedForUserID = jsonBody.get(JsonKey.REQUEST).get(JsonKey.USER_ID).asText();
+      }
+    } else { // for read-api
+      String uuidSegment = null;
+      Path path = Paths.get(request.uri());
+      if (request.queryString().isEmpty()) {
+        uuidSegment = path.getFileName().toString();
+      } else {
+        String[] queryPath = path.getFileName().toString().split("\\?");
+        uuidSegment = queryPath[0];
+      }
+      try {
+        requestedForUserID = UUID.fromString(uuidSegment).toString();
+      } catch (IllegalArgumentException iae) {
+        logger.error("Perhaps this is another API, like search that doesn't carry user id.");
+      }
+    }
+    return requestedForUserID;
+  }
+
+  /**
+   * Authenticates given HTTP request context
+   *
+   * @param request HTTP play request
+   * @return User or Client ID for authenticated request. For unauthenticated requests, UNAUTHORIZED
+   *     is returned release-3.0.0 on-wards validating managedBy token.
+   */
+  public static String verifyRequestData(Http.Request request) {
+    String clientId = JsonKey.UNAUTHORIZED;
+    request.flash().put(JsonKey.MANAGED_FOR, null);
+    Optional<String> accessToken = request.header(HeaderParam.X_Authenticated_User_Token.getName());
+    Optional<String> authClientId = request.header(HeaderParam.X_Authenticated_Client_Id.getName());
+      // The API must be invoked with either access token or client token.
+      if (accessToken.isPresent()) {
```

Comment:
@Hari-stackroute : what will happen if access token is not sent in the API header? whole authentication will be by-passed.

Reply:
@vrayulu : access token is mandatory to accessing the groups api's, there is no private api's provided here. so response will be sent as UNAUTHORISED.

Reply:
@Hari-stackroute - I recommend we should still have an ignore list similar to how we have in user API (and the list can be empty for group service now). the reason is that we will not need any code change when we add APIs that dont require access token check.